spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationManagementConfigurer.java

```diff
@@ -246,7 +247,8 @@ public void afterSingletonsInstantiated() {
 			this.metricsFactory = new DefaultMetricsFactory();
 		}
 		this.sourceConfigurers.putAll(this.applicationContext.getBeansOfType(MessageSourceMetricsConfigurer.class));
-		Map<String, IntegrationManagement> managed = this.applicationContext.getBeansOfType(IntegrationManagement.class);
+		Map<String, IntegrationManagement> managed = this.applicationContext
+				.getBeansOfType(IntegrationManagement.class);
```

Comment:
Shouldn't we suppress eager init here too? (and other gBOT calls in this class).

Reply:
Well, I believe it isn't going to harm, but at the same time it is idempotent here: all `getBeansOfType()` is called only from the ` afterSingletonsInstantiated()` which is done only once.